src/main/scripts/src/app/core/rest/collection.service.ts

```diff
@@ -19,27 +19,34 @@
  */
 
 import {Injectable} from '@angular/core';
-import {HttpClient} from './http-client.service';
+import {HttpClient, HttpParams} from '@angular/common/http';
+
 import {WorkspaceService} from '../workspace.service';
 import {Observable} from 'rxjs/Observable';
-import {isUndefined} from 'util';
+import {Collection} from '../dto/collection';
 
 @Injectable()
 export class CollectionService {
 
   constructor(private http: HttpClient, private workspaceService: WorkspaceService) {
   }
 
-  public getCollections(pageNumber?: number, pageSize?: number): Observable<any[]> {
-    let queryParams = !isUndefined(pageNumber) && !isUndefined(pageSize) ? `?page=${pageNumber}&size=${pageSize}` : '';
+  public getCollections(pageNumber?: number, pageSize?: number): Observable<Collection[]> {
+    let queryParams;
```

Comment:
You can assign `new HttpParams()` value here and use builder methods on `queryParams` in the block below.